src/files.ts

```diff
@@ -1,37 +1,34 @@
-import path from "path";
 import fs from "fs";
+import path from "path";
+
+/**
+ * A deep search function that resolves all files within a directory based on a list of extensions.
+ *
+ * @param {string} pathLike The base path to search through.
+ * @param {string[]} extensions The array of extensions to search for.
+ * @returns A list of paths for all the valid files within the given directory.
+ */
+
+export const resolveFiles = (pathLike: fs.PathLike, extensions: string[]): string[] => {
+	let results: string[] = [];
+
+	// collect directory contents
+	const fileList = fs.readdirSync(pathLike);
+
+	fileList.forEach((file: string, _) => {
```

Comment:
`_` is not needed here, `forEach` works with any function that takes `0` to `3` arguments. (see [`Array.prototype.forEach()`][mdn-foreach])

[mdn-foreach]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach "via MDN Web Docs"